components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -803,26 +803,29 @@ private String getMessageAsString(MessageConsumer messageConsumer) throws AxisFa
      * Pops the message from the associated queue.
      *
      * @param processorName message processor name.
+     * @return <code>true</code> if ++ is successful, else <code>false</code>
      */
-    public void popMessage(String processorName) throws AxisFault {
+    public boolean popMessage(String processorName) throws AxisFault {
         SynapseConfiguration configuration = getSynapseConfiguration();
         MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
 
         try {
-            popMessageFromQueue(messageConsumer);
+            return popMessageFromQueue(messageConsumer);
         } catch (SynapseException e) {
             handleException(log, "Failed to pop message from the queue: ", e);
+            return false;
         } finally {
             messageConsumer.cleanup();
         }
     }
 
-    private  void popMessageFromQueue(MessageConsumer messageConsumer) throws AxisFault {
+    private  boolean popMessageFromQueue(MessageConsumer messageConsumer) throws AxisFault {
         try {
             messageConsumer.receive();
-            messageConsumer.ack();
+            return messageConsumer.ack();
```

Comment:
In popAndRedirectMessageToStore we return false if a message is not received. Shall we do the same here?

Reply:
I will look into this and see. Please give me a moment.

Reply:
Yes. I think it will be safer to add a check to see if `messageConsumer.recieve()` actually gets a messageContext. 

Without the null check for the received messageContext, it will try to send an ack() to a null and fail, and the message wont get popped. Ill make the changes.